### DISCUSSION TITLE
Port Fowarding on BT Router

### DISCUSSION THREAD

**Discussion by AlanRichey** [GerryDazoo/Slinger#182] [Q&A]
I have installed the program with help from a user who has it working.   But he does not use a BT router so is unable to help with the port forwarding.    He recommended I change the default server port in CONFIG.INI to 65432 to avoid any clashes.

In the BT router the port forwarding page says this:

Protocol(Any,TCP,UDP)      Port Range   xxxxx  xxxxx        Translate to   xxxxx   xxxxx

Can someone tell me what to put in the boxes ?     If needed my Slingbox is on port 5301

Thanks

**Comment by floormanager1** [comment]
Hi, it's quite straightforward. The port range is the port you are going to use,  5301.  You can put that in all the port boxes - like in this video
[https://www.youtube.com/watch?v=sgFLQE4s2Z4](url)

You will need to open a port to your server, if you want to connect to your slingbox outside your network.  I opened two ports, one for the Slingbox and one for the server (which is at home).

**Reply by AlanRichey** [reply]
Thanks for the reply, but it left me even more confused.   In CONFIG.INI I have set my server port to 65432 and my Slingbox uses Port 5301, and you talk about opening a second port ?     So I'm afaid I am still none the wiser as to what numbers I put in the 4 boxes.

**Comment by floormanager1** [comment]
Okay, for the slingbox, find it's IP address.  Create a port-forwarding rule for it and in all 4 boxes, enter 5301.  Select TCP/UDP.
For the server, find the IP address.  Add another port-forwarding rule for that IP address by adding 65432 in all 4 boxes and again select TCP/UDP.  You should have now opened 2 ports, one for the Slingbox and one for the server.  You might want to force the router to always use the IP addresses for the slingbox and router, so that it does not allocate another IP address.

This is a useful tool for seeing if you've opened the ports
[https://canyouseeme.org/](url)

**Reply by GerryDazoo** [reply]
You don't need to open any ports if you don't plan to use the server or the slingbox remotely.
If you need remote access to the server open [SERVER] port (65432).
If you need remote access to the slingbox open the slingbox port i.e. 5301.
You only need to use the TCP protocol.  But if you do configure Both TCP/UDP it shouldn't hurt.

**Reply by surveyor2** [reply]
I have forwarding set on the server port 65432 only. That gives access to the sling server. I thought the slings used upnp to open whatever ports they need to connect back to the clients. It's working okay with remote clients anyway just with that one port forwarded.

James

**Comment by Pthepeg** [comment]
If your slingbox_server.py is running on a host on the same LAN as your slingbox, you only need to open one port, to the slingbox_server host. That would be Port Range 65432 65432 Translate to 65432 65432.... for the IP address of the slingbox_server host. That allows you to access the slingbox_server from some remote location outside your LAN.

**Comment by AlanRichey** [comment]
Thanks everyone, it's much clearer now.   Historically I have only used port forwarding ro do what it says, i.e. send one port to another.   Now I see I have to put the same in each port to open it.

I can now happily run my Slingox locally by entering  http://192.168.1.218/slingbox in 'Open Network Stream' in VLC.    I just need to figure out what to enter to watch it remotely.   I know that will be in the instructions somewhere so I will look there :-)

**Comment by floormanager1** [comment]
Glad you've got there.  I leave a server on at home, so I have a port open for the server.  In your case this will be 65432.  You can have your server outside the home but again you'd have to open a port, presumably not easy if you're in a hotel.  I use Google remote desktop to access the server, so I can turn on or off the server away from home.

To watch, you need your external IP address.  To make life easer for me, I have set up a DNS with https://www.noip.com/ so I do not have to worry if the IP address changes after a power outage etc.  So to connect to the slingbox, I have the server running and wherever I am, I connect using http://mynoipaddress:63000/slingbox

I'm not a computer expert or coder, just someone who's trying to keep access to my slingbox after 9th November.  There may be easier ways to do what I do but it works for me.
Paul
